ngi_reports/utils/entities.py

```diff
@@ -0,0 +1,453 @@
+    """Aborted Sample info class
+    """
+    def __init__(self, user_id, status):
+        self.status = status
+        self.user_id = user_id
+
+class Project:
+    """Project class
+    """
+    def __init__(self):
+        self.aborted_samples = OrderedDict()
+        self.samples = OrderedDict()
+        self.flowcells = {}
+        self.accredited = { 'library_preparation': 'N/A',
+                            'data_processing': 'N/A',
+                            'sequencing': 'N/A',
+                            'data_analysis':'N/A'
+                            }
+        self.application = ''
+        self.best_practice = False
+        self.cluster = ''
+        self.contact = ''
+        self.dates = { 'order_received': None,
+                        'open_date':  None,
+                        'first_initial_qc_start_date': None,
+                        'contract_received': None,
+                        'samples_received': None,
+                        'queue_date': None,
+                        'all_samples_sequenced': None
+                    }
+        self.is_finished_lib = False
+        self.is_hiseqx = False
+        self.library_construction  = ''
+        self.missing_fc  = False
+        self.ngi_facility = ''
+        self.ngi_name = ''
+        self.not_as_million  = False
+        self.num_samples = 0
+        self.num_lanes = 0
+        self.ngi_id = ''
+        self.reference = { 'genome': None,
+                            'organism': None }
+        self.report_date = ''
+        self.sequencing_setup = ''
+        self.skip_fastq = False
+        self.user_ID = ''
+
+    def populate(self, log, organism_names, **kwargs):
+
+        project = kwargs.get('project', '')
+        if not project:
+            raise SystemExit()
+        self.skip_fastq = kwargs.get('skip_fastq')
+        self.cluster = kwargs.get('cluster')
+        self.not_as_million = kwargs.get('not_as_million')
+
+        pcon = statusdb.ProjectSummaryConnection()
+        assert pcon, 'Could not connect to {} database in StatusDB'.format('project')
+
+        if re.match('^P\d+$', project):
+            self.ngi_id = project
+            id_view, pid_as_uppmax_dest = (True, True)
+        else:
+            self.ngi_name = project
+            id_view, pid_as_uppmax_dest = (False, False)
+
+        proj = pcon.get_entry(project, use_id_view=id_view)
+        if not proj:
+            log.error('No such project name/id "{}", check if provided information is right'.format(project))
+            raise KeyError
+        self.ngi_name = proj.get('project_name')
+
+        if proj.get('source') != 'lims':
+            log.error('The source for data for project {} is not LIMS.'.format(project))
+            raise BaseException
+
+        proj_details = proj.get('details',{})
+
+        if 'aborted' in proj_details:
+            log.warn('Project {} was aborted, so not proceeding.'.format(project))
+            raise SystemExit
```

Comment:
See above, I think we should get rid of the silent SystemExit

Reply:
Done!